File: SPOJ/FCTRL2_Small factorials/24_Small factorials.R
# Project name : SPOJ: Sum of digits
# Author       : Wojciech Raszka
# Date created : 2019-02-03
# Description  :
# Status       : Accepted (???)
# Comment      :

library(gmp, warn.conflicts=FALSE)

f <- file('stdin', open='r')

N = readLines(f, n=-1, warn=FALSE)

for (n in N){
  write(as.character(2*as.bigq(n) - 2), stdout())
}

File: SPOJ/FCTRL2_Small factorials/24_Small factorials.py
# Project name : SPOJ: FCTRL2 - Small factorials
# Author       : Wojciech Raszka
# E-mail       : contact@gitpistachio.com
# Date created : 
# Description  :
# Status       : 
# Tags         : python
# Comment      :

def factorial(n):
    if n > 1:
        return n*factorial(n - 1)

    return 1

test_cases = int(input())
numbers = [int(input()) for tc in range(test_cases)]

for n in numbers:
    print(factorial(n))
